lib/improver/tests/spotdata/test_build_spotdata_cube.py

```diff
@@ -0,0 +1,166 @@
+        self.assertEqual(result.units, 'degC')
+
+        # check coordinate values and units
+        self.assertEqual(result.coord('spot_index').points[0], 0)
+        self.assertAlmostEqual(result.coord('altitude').points[0], 10.)
+        self.assertEqual(result.coord('altitude').units, 'm')
+        self.assertAlmostEqual(result.coord('latitude').points[0], 59.5)
+        self.assertEqual(result.coord('latitude').units, 'degrees')
+        self.assertAlmostEqual(result.coord('longitude').points[0], 1.3)
+        self.assertEqual(result.coord('longitude').units, 'degrees')
+        self.assertEqual(result.coord('wmo_id').points[0], '03854')
+
+    def test_site_list(self):
+        """Test output for a list of sites"""
+        data = np.array([1.6, 1.3, 1.4, 1.1])
+        result = build_spotdata_cube(
+            data, 'air_temperature', 'degC', self.altitude, self.latitude,
+            self.longitude, self.wmo_id)
+
+        self.assertArrayAlmostEqual(result.data, data)
+        self.assertArrayAlmostEqual(
+            result.coord('altitude').points, self.altitude)
+        self.assertArrayAlmostEqual(
+            result.coord('latitude').points, self.latitude)
+        self.assertArrayAlmostEqual(
+            result.coord('longitude').points, self.longitude)
+        self.assertArrayEqual(result.coord('wmo_id').points, self.wmo_id)
+
+    def test_neighbour_method(self):
+        """Test output where neighbour_methods is populated"""
+        data = np.array([[1.6, 1.7], [1.3, 1.5], [1.4, 1.4], [1.1, 1.3]])
+
+        result = build_spotdata_cube(
+            data, 'air_temperature', 'degC', self.altitude, self.latitude,
+            self.longitude, self.wmo_id,
+            neighbour_methods=self.neighbour_methods)
+
+        self.assertArrayAlmostEqual(result.data, data)
+        self.assertEqual(result.coord_dims('neighbour_selection_method')[0], 1)
+        self.assertArrayEqual(
+            result.coord('neighbour_selection_method').points, np.arange(2))
+        self.assertArrayEqual(
+            result.coord('neighbour_selection_method_name').points,
+            self.neighbour_methods)
+
+    def test_grid_attributes(self):
+        """Test output where grid_attributes is populated"""
+        data = np.array([[1.6, 1.7, 1.8], [1.3, 1.5, 1.5],
+                         [1.4, 1.4, 1.5], [1.1, 1.3, 1.4]])
+
+        result = build_spotdata_cube(
+            data, 'air_temperature', 'degC', self.altitude, self.latitude,
+            self.longitude, self.wmo_id, grid_attributes=self.grid_attributes,
+            grid_attributes_dim=1)
+
+        self.assertArrayAlmostEqual(result.data, data)
+        self.assertEqual(result.coord_dims('grid_attributes')[0], 1)
+        self.assertArrayEqual(
+            result.coord('grid_attributes').points, np.arange(3))
+        self.assertArrayEqual(
+            result.coord('grid_attributes_key').points, self.grid_attributes)
+
+    def test_3d_spot_cube(self):
```

Comment:
I think it might be a good idea to have a test similar to this, but where the data dimensions doesn't match up to the lengths of the coordinates you are adding. I think the error raised by Iris in this case is probably ok, but it would be nice to see it tested.

Reply:
Test added.